packages/network-controller/tests/provider-api-tests/helpers.ts

```diff
@@ -400,10 +399,19 @@ export const withNetworkClient = async (
       return Promise.resolve();
     });
 
+  const networkConfigurationId = controller.upsertNetworkConfiguration(
+    {
+      rpcUrl: customRpcUrl,
+      chainId: '0x9999',
+      ticker: 'TEST',
+    },
+    { referrer: 'https://test-dapp.com', source: 'dapp' },
+  );
+
   if (providerType === 'infura') {
     controller.setProviderType(infuraNetwork);
-  } else {
-    controller.setRpcTarget(customRpcUrl, customChainId);
+  } else if (networkConfigurationId) {
```

Comment:
Nit: Maybe we should throw if neither is set? Assuming that either one or the other is expected.

Reply:
modified this slightly to make more sense: https://github.com/MetaMask/core/pull/1064/commits/0124a625a57d81ae00c1dcaa03e880c62f6ff14e#diff-feb9bf14adef7e507b7979d14206a62ebbecc1842a65802807fa31aefd566fbbR404-R412